### PR TITLE
feat: clicking on text or icon in file view will now download/open the file

### PR DESCRIPTION
**Pull request by Karim94**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/2795/mobile-in-file-preview-screen-file-icon-and-name-should-be-tappable
#### Testing instructions  

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: app/components/files/file-view.js [hunk 1]
```diff
@@ -60,17 +77,21 @@ export default class FileView extends SafeComponent {
...
+                                        type={file.iconType}
+                                    />}
+                            </TouchableOpacity>
                         </View>
                         <View style={firstColumnStyle}>
                             <View style={{ flexGrow: 1, flexShrink: 1 }}>
-                                <Text
-                                    ellipsizeMode="tail"
-                                    numberOfLines={1}>{file.name}</Text>
+                                <TouchableOpacity onPress={() => this.fileAction(file)}>
```
Comment (seavan) [inline app/components/files/file-view.js]: arrow function should be @action.bound

File: app/components/files/file-view.js [hunk 2]
```diff
@@ -42,6 +42,23 @@ export default class FileView extends SafeComponent {
         return <FileActions file={this.file} />;
     }
 
+    openFile(file) {
+        return Promise.resolve()
+            .then(() => file.launchViewer())
+            .finally(() => { uiState.externalViewer = false; });
```
Comment (seavan) [inline app/components/files/file-view.js]: this code is copy-pasted from FileActions. it's outdated and doesn't look nice.
also copy-pasting is not great.

after analyzing the code i see that uiState.externalViewer is not used anymore properly. please remove it and replace openFile(file) with file.launchViewer().

File: app/components/files/file-view.js [hunk 3]
```diff
@@ -42,6 +42,23 @@ export default class FileView extends SafeComponent {
...
+    }
+
+    fileAction(file) {
```
Comment (seavan) [inline app/components/files/file-view.js]: @action.bound

File: app/components/files/file-view.js [hunk 4]
```diff
@@ -42,6 +42,23 @@ export default class FileView extends SafeComponent {
...
+        const enabled = file && file.readyForDownload || fileState.showSelection;
+        if (enabled) {
+            if (file && !file.isPartialDownload && file.cached) {
+                this.openFile(file);
+            } else {
+                fileState.download();
```
Comment (seavan) [inline app/components/files/file-view.js]: file.launchViewer()

and

fileState.download(file)

Comment (Karim94) [reply app/components/files/file-view.js]: Why do we need to specify the file? Doesn't it use filestate.currentFile ?

Comment (seavan) [reply app/components/files/file-view.js]: I don't like the ambiguity. I feel that this is not very clear and may be source of errors in the future.

**Comment by seavan** [comment]
@Karim94 bump

**Comment by seavan** [comment]
@ebilg ready for testing

**Comment by ebilg** [comment]
can be merged!

File: app/components/files/file-view.js [hunk 5]
```diff
@@ -61,17 +74,21 @@ export default class FileView extends SafeComponent {
                             marginLeft: vars.spacing.small.mini2x,
                             marginRight: vars.spacing.small.maxi2x
                         }}>
-                            {icon ||
+                            <TouchableOpacity onPress={this.fileAction}>
```
Comment (seavan) [inline app/components/files/file-view.js]: needs pressRetentionOffset, @Karim94

File: app/components/files/file-view.js [hunk 6]
```diff
@@ -61,17 +74,21 @@ export default class FileView extends SafeComponent {
...
                                     type={fileHelpers.getFileIconType(file.ext)}
                                 />}
+                            </TouchableOpacity>
                         </View>
                         <View style={firstColumnStyle}>
                             <View style={{ flexGrow: 1, flexShrink: 1 }}>
-                                <Text
-                                    ellipsizeMode="tail"
-                                    numberOfLines={1}>{file.name}</Text>
+                                <TouchableOpacity onPress={this.fileAction}>
```
Comment (seavan) [inline app/components/files/file-view.js]: needs pressRetentionOffset, @Karim94

**Pull request merged**